libraries/db/undo_database.cpp

```diff
@@ -118,15 +118,19 @@ void undo_database::undo()
       _db.insert( std::move(*item.second) );
 
    _stack.pop_back();
-   if( _stack.empty() )
-      _stack.emplace_back();
```

Comment:
curious why we had this in the first place, and why we think these two lines should be removed?

Reply:
I think these two lines were added as a workaround for the problem I describe here: https://github.com/bitshares/bitshares-core/pull/339#issuecomment-320519955

https://github.com/bitshares/bitshares-core/pull/339/commits/4822c210a3f714a28c2c05487e03bf1202b7865f contains a better fix to the underlying problem, IMO.

Reply:
not sure I understand all the corner cases, but it looks better. thanks.